src/Command/Branch/BranchChangelogCommand.php

```diff
@@ -48,41 +46,51 @@ protected function execute(InputInterface $input, OutputInterface $output)
         try {
             $latestTag = $this->getHelper('git')->getLastTagOnBranch();
         } catch (\RuntimeException $e) {
-            $output->writeln('<info>There were no tags found</info>');
+            $this->getHelper('gush_style')->note(
+                sprintf('No tags were found on branch "%s".', $this->getHelper('git')->getActiveBranchName())
+            );
 
             return self::COMMAND_SUCCESS;
         }
 
         $adapter = $this->getIssueTracker();
         $commits = $this->getHelper('git')->getLogBetweenCommits($latestTag, 'HEAD');
+        $issues = $this->getIssuesFromCommits($commits);
+
+        foreach ($issues as $id) {
+            // ignore missing issues
+            try {
+                $issue = $adapter->getIssue($id);
+            } catch (\Exception $e) {
+                continue;
+            }
+
+            $output->writeln(
+                sprintf("#%s: %s   <info>%s</info>", $id, $issue['title'], $issue['url'])
+            );
+        }
+
+        return self::COMMAND_SUCCESS;
+    }
 
-        // Filter commits that reference an issue
+    /**
+     * @param array $commits
+     *
+     * @return integer[]
+     */
+    private function getIssuesFromCommits(array $commits)
+    {
         $issues = [];
 
         foreach ($commits as $commit) {
-            // Cut issue id from branch name (merge commits)
-            if (preg_match('/\/#([0-9]+)/i', $commit['message'], $matchesGush) && isset($matchesGush[1])) {
+            if (preg_match('{#([0-9]+)}i', $commit['message'], $matchesGush)) {
```

Comment:
This check is very strict, and won't match for example Jira issues. I think each issue tracker adapter should have it's own `getIssueRegex` (or something more creative :)) method so that you can match a issue better based on the adapter